### ISSUE TITLE
enhancement request: support projected partitions in Athena

### ISSUE THREAD

**Issue by kub** [localstack/localstack#9514]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Context:
Athena currently supports partition projection, which allows for dynamically discovering and projecting partitions for a table without the need to run `MSCK REPAIR TABLE` or similar operations. This feature can be particularly useful for tables with a large number of partitions, as it can significantly speed up query performance by avoiding unnecessary metadata calls.

### Proposal:
Implement support for projected partitions in the Athena implementation within LocalStack.
Allow users to define partition projection configurations, including the source of projection (e.g., a range of dates) and the projection type (e.g., enumerated, injected, etc.). Ensure that the projected partitions are recognized and utilized during query execution to optimize performance.

**Comment by kub** [comment]
is there any ETA for this feature?

**Comment by alexrashed** [comment]
No, unfortunately there's no ETA for this feature yet. I just re-opened it because it's a valid feature request and should not be closed by the stale bot due to inactivity.

**Comment by HarshCasper** [comment]
Due to the forthcoming changes in our distribution and development model ([read more](https://blog.localstack.cloud/the-road-ahead-for-localstack/)), this issue tracker is no longer being maintained. We've migrated this issue to our internal tracker and will continue to work on it there. If you'd like to follow up on the status or expedite this, feel free to reach out through one of our support channels:

-   💬 [GitHub Discussions](https://github.com/orgs/localstack/discussions)
-   📖 [Help & Support](https://docs.localstack.cloud/aws/getting-started/help-support/)
